terraform/aptos-node/gcp/README.md

```diff
@@ -0,0 +1,180 @@
+  # This command will list all workspaces
+  $ terraform workspace list
+  ```
+
+7. Apply the configuration.
+
+  ```
+  $ terraform apply
+  ```
+
+  This might take a while to finish (10 - 20 minutes), Terraform will create all the resources on your cloud account. 
+
+8. Once terraform apply finishes, you can check if those resources are created:
+
+    - `kubectl get pods` this should have haproxy, validator and fullnode. with validator and fullnode pod `pending` (require further action in later steps)
+    - `kubectl get svc` this should have `validator-lb` and `fullnode-lb`, with an external-IP you can share later for connectivity.
+
+9. Get your node IP info:
+
+    ```
+    $ export VALIDATOR_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-validator-lb --output jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+
+    $ export FULLNODE_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-fullnode-lb --output jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+    ```
+
+10. Generate key pairs (node owner key, consensus key and networking key) in your working directory.
+
+    ```
+    $ aptos genesis generate-keys --output-dir ~/$WORKSPACE
+    ```
+
+    This will create a `private-keys.yaml` file for you, Backup your key files somewhere safe, this is important for you to establish ownership of your node, and it will be used to claim your rewards later if eligible. Very important!!
+
+11. Configure validator information.
+
+    ```
+    $ aptos genesis set-validator-configuration --keys-dir ~/$WORKSPACE --local-repository-dir ~/$WORKSPACE --username <pick a username for your node> --validator-host $VALIDATOR_ADDRESS:6180 --full-node-host $FULLNODE_ADDRESS:6182
+
+    ```
+
+    This will create a YAML file in your working directory with your username, e.g. `aptosbot.yml`, it should looks like:
+
+    ```
+    ---
+    account_address: 7410973313fd0b5c69560fd8cd9c4aaeef873f869d292d1bb94b1872e737d64f
+    consensus_key: "0x4e6323a4692866d54316f3b08493f161746fda4daaacb6f0a04ec36b6160fdce"
+    account_key: "0x83f090aee4525052f3b504805c2a0b1d37553d611129289ede2fc9ca5f6aed3c"
+    network_key: "0xa06381a17b090b8db5ffef97c6e861baad94a1b0e3210e6309de84c15337811d"
+    validator_host:
+      host: 35.232.235.205
+      port: 6180
+    full_node_host:
+      host: 34.135.169.144
+      port: 6182
+    stake_amount: 1
+    ```
+
+12. Create layout YAML file, which defines the node in the validatorSet, for test mode, we can create a genesis blob containing only one node.
+
+    ```
+    $ vi layout.yml
+    ```
+
+    Add root key, node username, and chain_id in the `layout.yml` file, for example:
+
+    ```
+    ---
+    root_key: "0x5243ca72b0766d9e9cbf2debf6153443b01a1e0e6d086c7ea206eaf6f8043956"
+    users:
+      - <username you created in step 5>
+    chain_id: 5
+    ```
+
+13. Download AptosFramework Move bytecodes into a folder named `framework`
+
+    ```
+    $ wget https://github.com/aptos-labs/aptos-core/files/8665561/framework.zip
```

Comment:
we should point them instead to the latest release on the Github releases page: https://github.com/aptos-labs/aptos-core/tags

e.g. https://github.com/aptos-labs/aptos-core/archive/refs/tags/aptos-framework-v.0.0.1.zip